torch/testing/_internal/opinfo_helper.py

```diff
@@ -0,0 +1,110 @@
+        # Return a map of (device, supported_dtypes)
+        device_dtype = defaultdict(list)
+        for device in DEVICES:
+            for dtype in all_types_and_complex_and(torch.bool, torch.bfloat16, torch.half):
+                sample = self.sample_inputs_fn(self.op, device, dtype, False)[0]
+                try:
+                    self.op(sample.input, *sample.args, **sample.kwargs)
+                    device_dtype[device].append(dtype)
+                except RuntimeError as re:
+                    if "not implemented for" not in str(re):
+                        raise re
+                    pass
+
+        return device_dtype
+
+    def has_out_support(self, device, dtype):
+        try:
+            sample = self.sample_inputs_fn(None, device, dtype, False)[0]
+            op_out = partial(self.op, sample.input, *sample.args, **sample.kwargs)
+            op_out(out=torch.empty_like(sample.input))
+            return True
+        except TypeError as te:
+            if "unexpected keyword argument 'out'" not in str(te):
+                raise te
+            return False
+
+    def dtype_dispatch_hint(self, dtypes):
+        # Find the closest dispatcher from the set of available dispatchers
+
+        # Metric for finding the closest dispatcher
+        # Intersection Over Union
+        def iou(dispatch_dtypes):
+            return len(set(dtypes).intersection(dispatch_dtypes)) * 1.0 / len(set(dtypes).union(dispatch_dtypes))
+
+        dispatch_iou_score = {dispatch: iou(dispatch()) for dispatch in DTYPE_DISPATCH}
+
+        dispatch, _ = max(dispatch_iou_score.items(), key=lambda x: x[1])
+
+        return dispatch.__name__ + " + " + str(tuple(set(dtypes) - set(dispatch())))
+
+    def summary(self):
+        device_supported_dtypes = self.get_supported_dtypes()
+        supports_out = self.has_out_support('cpu', device_supported_dtypes['cpu'][0])
+
+        # `dtypes` argument of OpInfo
+        cpu_dtypes = device_supported_dtypes["cpu"]
+        cpu_dispatch_hint = self.dtype_dispatch_hint(cpu_dtypes)
+
+        # `dtypesIfCUDA` argument of OpInfo
+        cuda_dtypes = device_supported_dtypes['cuda']
+        cuda_dispatch_hint = self.dtype_dispatch_hint(cuda_dtypes)
+
+        summary_str = (f'OpInfo({self.op.name}',
+                       f'# hint: {cpu_dispatch_hint}',
+                       f'dtypes={cpu_dtypes}',
+                       f'# hint: {cuda_dispatch_hint}' if (cpu_dtypes != cuda_dtypes and TEST_CUDA) else '',
+                       f'dtypesIfCUDA={cuda_dtypes}' if (cpu_dtypes != cuda_dtypes and TEST_CUDA) else '',
+                       f'supports_out={supports_out}' if not supports_out else '',  # as supports_out is True by default
+                       f'sample_inputs_func={self.sample_inputs_fn.__name__})')
+
+        empty_spaces = " " * len('OpInfo(')
+        return (",\n" + empty_spaces).join(filter(None, summary_str))
+
+
+# Run using `python -m torch.testing._internal.opinfo_helper`
```

Comment:
This is cool that the file is runnable and suggests the correct thing to do. Maybe for now it can just suggest dtype corrections, and this is where the warning can go about lacking CUDA support?

Reply:
With your latest update to the test suite, is this being runnable still that interesting? Maybe we should remove this feature for now so we have more flexibility if we decide to add it later?